internal/report/report_test.go

```diff
@@ -1486,6 +1610,91 @@ func TestNewWriter_OutputFile(t *testing.T) {
 	}
 }
 
+func TestWriter_getStaleExclusions(t *testing.T) {
+	tests := []struct {
+		name       string
+		exclusions []config.Exclusion
+		vulns      []vulnerability
+		want       []config.Exclusion
+	}{
+		{
+			name: "without stale exclusions",
+			exclusions: []config.Exclusion{
+				{Summary: "Summary 1"},
+				{Resource: "Resource 1"},
+			},
+			vulns: []vulnerability{
+				{
+					Vulnerability: vreport.Vulnerability{
+						Summary:          "Vulnerability Summary 1",
+						Score:            6.7,
+						AffectedResource: "Resource 1",
+					},
+					matchedExclusions: []int{0, 1},
+				},
+			},
+			want: []config.Exclusion{},
+		},
+		{
+			name: "matched all exclusion in different vulnerabilities",
+			exclusions: []config.Exclusion{
+				{Summary: "Summary 1"},
+				{Resource: "Resource 2"},
+			},
+			vulns: []vulnerability{
+				{
+					Vulnerability: vreport.Vulnerability{
+						Summary:          "Vulnerability Summary 1",
+						Score:            6.7,
+						AffectedResource: "Resource 1",
+					},
+					matchedExclusions: []int{0, 1},
+				},
+				{
+					Vulnerability: vreport.Vulnerability{
+						Summary:          "Vulnerability Summary 2",
+						Score:            6.7,
+						AffectedResource: "Resource 1",
+					},
+					matchedExclusions: []int{0, 1},
+				},
+			},
```

Comment:
Given that `Writer.getStaleExclusions` do not care about the other `vulnerability` fields, I'd strip them:

```suggestion
			vulns: []vulnerability{
				{ matchedExclusions: []int{0, 1} },
				{ matchedExclusions: []int{0, 1} },
			},
```

Same for the other cases.